client/src/main/java/org/eclipse/hono/client/impl/AbstractHonoClient.java

```diff
@@ -249,8 +249,29 @@ protected static final void setApplicationProperties(final Message msg, final Ma
             receiver.setAutoAccept(true);
             receiver.setQoS(qos);
             receiver.setPrefetch(clientConfig.getInitialCredits());
-            receiver.handler(messageHandler);
-            receiver.openHandler(result.completer());
+            receiver.handler((delivery, message) -> {
+                messageHandler.handle(delivery, message);
+                if (LOG.isTraceEnabled()) {
+                    int remainingCredits = receiver.getCredit() - receiver.getQueued();
+                    LOG.trace("handling message [remotely settled: {}, queued messages: {}, remaining credit: {}]", delivery.remotelySettled(), receiver.getQueued(), remainingCredits);
+                }
+            });
+            receiver.openHandler(openAttach -> {
```

Comment:
so, it looks like there hasn't be a problem in the first place because a failed attach frame from the peer had already been forwarded to the result handler.
Anyway, removing duplicate code is always good :-)